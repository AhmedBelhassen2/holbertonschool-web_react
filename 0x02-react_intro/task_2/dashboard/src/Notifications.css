  
  .Notifications {
    border: thin dotted #e0344a;
    padding: 16px;
  }

  .Notifications ul {
    color: #e01d3f;
  }
  [data-priority="default"] {
    color: blue;
  }