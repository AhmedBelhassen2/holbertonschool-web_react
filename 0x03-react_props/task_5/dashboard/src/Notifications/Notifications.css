  
  .Notifications {
    border: 1px dashed #e0344a;
    padding: 16px;
  }

  .Notifications ul {
    color: #e01d3f;
  }
  [data-notification-type="default"]  {
    color: blue;
  }