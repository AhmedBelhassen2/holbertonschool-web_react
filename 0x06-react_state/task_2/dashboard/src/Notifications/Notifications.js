import React, { Component, Fragment } from 'react';
import closeIcon from '../assets/close-icon.png';
import NotificationItem from './NotificationItem';
import PropTypes from 'prop-types';
import NotificationItemShape from './NotificationItemShape';
import { StyleSheet, css } from 'aphrodite';

class Notifications extends Component {
	constructor(props) {
		super(props);
		this.markAsRead = this.markAsRead.bind(this);
	};

	shouldComponentUpdate(nextProps) {
		if (
			this.props.listNotifications.length < nextProps.listNotifications.length ||
			this.props.displayDrawer !== nextProps.displayDrawer
		) {
			return true;
		}
		return false;
	};

	markAsRead(id) {
		console.log(`Notification ${id} has been marked as read`);
	};

	render() {
		let {
			displayDrawer,
			listNotifications,
			handleDisplayDrawer,
			handleHideDrawer
		} = this.props;

		return (
			<div className="NotificationsComponent">
				<div
					className={`menuItem ${css(styles.menuItem)}`}
					onClick={() => handleDisplayDrawer()}
				>
					Your notifications
				</div>
				{
					displayDrawer &&
					<div className={css(styles.notifications)}>
						<button
							style={{
								color: '#3a3a3a',
								fontWeight: 'bold',
								background: 'none',
								border: 'none',
								fontSize: '15px',
								position: 'absolute',
								right: '3px',
								top: '3px',
								cursor: 'pointer',
								outline: 'none',
							}}
							aria-label="Close"
							onClick={(e) => {
								// console.log('Close button has been clicked');
								handleHideDrawer()
							}}
						>
							<img
								src={closeIcon}
								alt="close icon"
							/>
						</button>
						{
							listNotifications.length === 0 &&
							<p>No new notification for now</p>
						}
						{
							listNotifications.length > 0 &&
							<Fragment>
								<p>
									Here is the list of notifications
								</p>
								<ul>
									{	
										listNotifications.map((notif) => {
											return (
												<NotificationItem
													key={notif.id}
													id={notif.id}
													type={notif.type}
													value={notif.value}
													html={notif.html}
													markAsRead={this.markAsRead}
												/>
											)
										})
									}
								</ul>
							</Fragment>
						}
					</div>
				}
			</div>
		);
	};
};

const cssVars = {
  mainColor: "#e01d3f",
};

const screenSize = {
  small: "@media screen and (max-width: 900px)",
};

const opacityKeyframes = {
  from: {
    opacity: 0.5,
  },

  to: {
    opacity: 1,
  },
};

const translateYKeyframes = {
  "0%": {
    transform: "translateY(0)",
  },

  "50%": {
    transform: "translateY(-5px)",
  },

  "75%": {
    transform: "translateY(5px)",
  },

  "100%": {
    transform: "translateY(0)",
  },
};

const borderKeyframes = {
  "0%": {
    border: `3px dashed deepSkyBlue`,
  },

  "100%": {
    border: `3px dashed ${cssVars.mainColor}`,
  },
};

const styles = StyleSheet.create({
  menuItem: {
    float: "right",
    ":hover": {
      cursor: "pointer",
      animationName: [opacityKeyframes, translateYKeyframes],
      animationDuration: "1s, 0.5s",
      animationIterationCount: 3,
    },
  },

  menuItemPNoShow: {
    marginRight: "8px",
    display: "none",
  },

  menuItemPShow: {
    marginRight: "8px",
  },

  notifications: {
    float: "right",
    // border: `3px dashed ${cssVars.mainColor}`,
    padding: "10px",
    marginBottom: "20px",
    animationName: [borderKeyframes],
    animationDuration: "0.8s",
    animationIterationCount: 1,
    animationFillMode: "forwards",
    ":hover": {
      border: `3px dashed deepSkyBlue`,
      // animationFillMode: "forwards",
    },
    [screenSize.small]: {
      float: "none",
      border: "none",
      listStyle: "none",
      padding: 0,
      fontSize: "20px",
      ":hover": {
        border: "none",
        // animationFillMode: "forwards",
      },
      position: "absolute",
      background: "white",
      height: "110vh",
      width: "100vw",
    },
  },

  notificationsButtonImage: {
    width: "14px",
  },

  notificationsP: {
    margin: 0,
    marginTop: "15px",
  },

  notificationsUL: {
    [screenSize.small]: {
      padding: 0,
    },
  },
});

// css(styles.globals);

Notifications.propTypes = {
	displayDrawer: PropTypes.bool,
	listNotifications: PropTypes.arrayOf(NotificationItemShape),
	displayDrawer: PropTypes.bool,
	handleDisplayDrawer: PropTypes.func,
	handleHideDrawer: PropTypes.func,
};

Notifications.defaultProps = {
	displayDrawer: false,
	listNotifications: [],
	displayDrawer: false,
};

export default Notifications;