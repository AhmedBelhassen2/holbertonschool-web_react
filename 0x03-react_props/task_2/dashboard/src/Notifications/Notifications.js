import React from "react";
import "./Notifications.css";
import { getLatestNotification } from "../utils/utils";
import closeIcon from "../assets/close-icon.png";
import NotificationItem from "./NotificationItem";

export function myFunction() {
    document.getElementById("demo").innerHTML = "Close button has been clicked";
  }

const Notifications = () => {
  return (
      <div className="Notifications">
      <button 
        style={{
          background: "transparent",
          border: "none",
          position: "absolute",
          right:"-600px" ,
          left: "-20px", top: 10,
        }}
        aria-label="close"
        onClick={myFunction}>Click me
      <img src={closeIcon} alt="close-icon"
         style={{
            right: 20,
            width : "150px",
            height : "150px",
        }}
    />
        
      </button>
      <p>Here is the list of notifications</p>
      <ul>
      <NotificationItem type='default' value='New course available' />
      <NotificationItem type='urgent' value='New resume available' />
      <NotificationItem type='urgent' html={{__html:getLatestNotification()}} />
      </ul>
      <p id = "demo"></p>

    </div>
  );
};

export default Notifications;