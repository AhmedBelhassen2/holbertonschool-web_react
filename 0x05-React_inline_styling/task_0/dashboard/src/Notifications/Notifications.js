import React from "react";
import "./Notifications.css";
import { getLatestNotification } from "../utils/utils";
import closeIcon from "../assets/close-icon.png";


export function myFunction() {
    document.getElementById("demo").innerHTML = "Close button has been clicked";
  }

class Notifications extends React.Component  {

  constructor(props) {
    super(props);
    this.markAsRead = this.markAsRead.bind(this);
  }
  markAsRead(id) {
    console.log(`Notification ${id} has been marked as read`);
  }
  render ()  {
    return(
      <div className="Notifications">
      <button 
        style={{
          background: "transparent",
          border: "none",
          position: "absolute",
          right:"-600px" ,
          left: "-20px", top: 10,
        }}
        aria-label="close"
        onClick={myFunction}>Click me
      <img src={closeIcon} alt="close-icon"
         style={{
            right: 20,
            width : "150px",
            height : "150px",
        }}
    />
        
      </button>
      <p>Here is the list of notifications</p>
      
      <ul>
        <li data-priority="default">New course available</li>
        <li data-priority="urgent">New resume available</li>
        <li data-priority="urgent" dangerouslySetInnerHTML={{__html: getLatestNotification() }} />
      </ul>
      <p id = "demo"></p>

    </div>
    )
}}




export default Notifications;