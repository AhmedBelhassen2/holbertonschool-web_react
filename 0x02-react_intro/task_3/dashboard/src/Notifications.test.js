import React from 'react';
import Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Notifications from "./Notifications";

Enzyme.configure({adapter: new Adapter()})


let wrapper;

beforeEach(() => {
  wrapper = shallow(<Notifications />);
});
describe("Testing the <Notifications /> Component", () => {
    it('render without crashing', () => {
      expect(wrapper.exists()).toEqual(true);
    });
  
    it('renders three list items', () => {
      expect(wrapper.find('li')).toHaveLength(3);
    })
  
    it('renders the <p>', () => {
      expect(wrapper.containsMatchingElement(<p>Here is the list of notifications</p>)).toBeTruthy();
      });
});