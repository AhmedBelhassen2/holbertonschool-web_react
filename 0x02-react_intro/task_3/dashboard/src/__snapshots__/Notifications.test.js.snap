// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing component <App /> render without crashing 1`] = `ShallowWrapper {}`;

exports[`testing component <Notifications /> render without crashing 1`] = `ShallowWrapper {}`;
